engine/src/main/java/org/hibernate/search/query/engine/impl/QueryHits.java

```diff
@@ -399,13 +399,20 @@ private FacetResult getFacetResultForLongRange(RangeFacetRequest<?> facetRequest
 	}
 
 	private ArrayList<Facet> updateStringFacets(DiscreteFacetRequest facetRequest, FacetMetadata facetMetadata) throws IOException {
+		ArrayList<Facet> facets = new ArrayList<>();
 		SortedSetDocValuesReaderState docValuesReaderState;
 		try {
 			docValuesReaderState = new DefaultSortedSetDocValuesReaderState( searcher.getIndexReader() );
 		}
 		catch (IllegalArgumentException e) {
-			// happens in case there are no facets at all configured for the matching documents
-			throw log.unknownFieldNameForFaceting( facetRequest.getFacetingName(), facetRequest.getFieldName() );
+			/*
+			 * Happens in case there are no facets at all stored in the matching documents.
+			 * But we know some the target field is correctly configured to generate facets,
+			 * because we managed to retrieve the FacetMetadata.
+			 * So we can safely return an empty list: the matching documents simply do not have
+			 * any value for this field.
+			 */
+			return facets;
```

Comment:
Can't we use `Collections.emptyList()` and keep the array list initialization where it was? I think the list is readonly after that? Let's not initialize an array list for the sake of it.

Reply:
Well the facet list we return to the user has always been mutable, and I'm dead certain some people must rely on it.
Plus the calling code mutates the list in some cases. Not when it's empty, granted, but still I would feel bad returning an immutable list.

So... Not in a minor?

Reply:
Right, I thought we would have made them immutable at some point.

Merged!